src/parser.rs

```diff
@@ -952,6 +953,29 @@ impl Library {
             "attribute" => parser.ignore_element(),
             _ => Err(parser.unexpected_element(elem)),
         })?;
+        let mut fix_pos = 0;
+        // Fixing the closure attributes in case they're badly set.
+        for index in 0..params.len() {
+            if params[index].instance_parameter {
+                fix_pos += 1;
+            } else if let Some(pos) = params[index].closure {
+                if params[pos - fix_pos].closure.is_none() {
```

Comment:
This is setting it so that callback points at user data and the other way around? The attribute is only supposed to be on the callback